pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -865,3 +871,21 @@ def build_agent_graph(
         auto_instrument=False,
     )
     return graph
+
+
+async def _process_message_history(
+    messages: list[_messages.ModelMessage],
+    processors: Sequence[HistoryProcessor],
+) -> list[_messages.ModelMessage]:
+    """Process message history through a sequence of processors."""
+    processed_messages = messages
+    for processor in processors:
+        if inspect.iscoroutinefunction(processor):
+            async_processor = cast(
+                Callable[[list[_messages.ModelMessage]], Awaitable[list[_messages.ModelMessage]]], processor
+            )
+            processed_messages = await async_processor(processed_messages)
```

Comment:
Note to future us: with Temporal, this will need to be done in an activity as it could include IO (like when the processor using a summarization agent).